keps/sig-api-machinery/2896-openapi-v3/README.md

```diff
@@ -226,31 +227,32 @@ proposal will be implemented, this is the place to discuss them. -->
 
 ### Paths
 
-The overarching `/openapi/v3` endpoint will contain the list of paths (groups)
+The root `/openapi/v3` endpoint will contain the list of paths (groups)
 available and serve as a discovery endpoint. Clients can then choose the
 group(s) to fetch and send additional requests.
 
 /openapi/v3
 
 ```json
 {
-   "Paths" : [
-      "api",
-      "api/v1",
-      "apis",
-      "apis/admissionregistration.k8s.io",
-      "apis/apiextensions.k8s.io",
-      "apis/apps",
-      "apis/authentication.k8s.io",
+   "Paths" : {
+      "api": "/openapi/v3/api?etag=tag",
+      "api/v1": "/openapi/v3/api/v1?etag=tag",
+      "apis": "/openapi/v3/apis?etag=tag",
+      "apis/admissionregistration.k8s.io/v1": "/openapi/v3/apis/admissionregistration.k8s.io/v1?etag=tag",
+      "apis/apiextensions.k8s.io/v1": "/openapi/v3/apis/apiextensions.k8s.io/v1?etag=tag",
+      "apis/apps/v1": "/openapi/v3/apis/apps/v1?etag=tag",
       ...
-   ]
+   }
 }
 ```
 
 Based on the provided group, clients can then request `openapi/v3/apis/apps/v1`,
 `/openapi/v3/apis/networking.k8s.io/v1` and etc. These leaf node specs are self
 contained OpenAPI v3 specs and include all referenced types.
 
+The discovery document has the format of a map with the key being the group-version and value representing the URL of the OpenAPI for the particular group version. Note that the URL can be constructed by the client by prepending the group-version name with the `openapi/v3` prefix. The URL listed here provides a special etag query parameter to denote the latest etag for the OpenAPI spec for the particular group-version. All OpenAPI spec requests with the `?etag` query parameter will return a response with `Cache Control: immutable`. This allows clients to cache the OpenAPI spec when the etag is not changed. Support for caching is built into the httpcache library used in client-go, and no change is needed on the client side to support this mechanism other than passing the additional query parameter. Passing the etag as the query parameter allows clients to check the etag in the root discovery document. Clients can avoid sending an additional request to fetch an ETag from specific group-versions, and the root document itself can provide information on group-version changes and updates. If there is a race and the client passes in an outdated etag value, the server will send a 301 to redirect the client to the URL with the latest etag value.
```

Comment:
Since we now know how this is called, mention that this is a super common pattern on the web called "cache busting".